prov/gni/src/gnix_vc.c

```diff
@@ -1661,27 +1656,28 @@ static int __gnix_vc_push_work_reqs(struct gnix_vc *vc)
 				  "Request processed: %p\n", req);
 			goto try_again;
 		} else {
-			/* Work failed.  Reschedule to put this VC back on the
-			 * end of the list. */
+			/* Work failed.  Reschedule to put this VC
+			 * back on the end of the list and return
+			 * -FI_EAGAIN */
 			__gnix_vc_work_schedule(vc);
 
 			fastlock_acquire(&vc->work_queue_lock);
 			slist_insert_tail(item, &vc->work_queue);
 			fastlock_release(&vc->work_queue_lock);
 
-			/* FI_ENOSPC is reserved to indicate a lack of TXDs,
-			 * which are shared by all VCs on the NIC.  Return
-			 * error to stall processing of VCs in this case.  The
-			 * other likely error is a lack of SMSG credits, which
-			 * only halts this VC. */
-			if (ret == -FI_ENOSPC) {
-				fi_rc = -FI_EAGAIN;
-			} else if (ret != -FI_EAGAIN) {
+			fi_rc = -FI_EAGAIN;
+
+			/* FI_ENOSPC is reserved to indicate a lack of
+			 * TXDs, which are shared by all VCs on the
+			 * NIC.  The other likely error is FI_EAGAIN
+			 * due to a lack of SMSG credits. */
+
+			if ((ret != -FI_ENOSPC) && (ret != -FI_EAGAIN)) {
```

Comment:
It only gets printed when it's *not* FI_ENOSPC and *not* FI_EAGAIN.  Or did I get the logic wrong?